actionpack/lib/action_controller/metal/request_forgery_protection.rb

```diff
@@ -253,6 +300,57 @@ def handle_unverified_request
       end
     end
 
+    module TokenStorageMethods
+      class Base
+        attr_reader :controller
+
+        def initialize(controller)
+          @controller = controller
+        end
+
+        private
+          def generate_csrf_token # :nodoc:
+            if controller.urlsafe_csrf_tokens
```

Comment:
I wonder if we could just get rid of this config. urlsafe tokens are the default in 6.1, and I see no use whatsoever for not wanting URLsafe token. I'll have to dig a bit.

Reply:
Ah, they're deprecated, so yeah we can just get rid of that: https://github.com/rails/rails/pull/43817

So ultimately I don't think this object should receive the `controller`. It should be instantiated only once, and then receive the request as argument of `retrieve_token`